samples/ClassroomLabs/Modules/Library/Tools/LabCreationLibrary.psm1

```diff
@@ -0,0 +1,598 @@
+            }
+
+            while (($jobs | Measure-Object).Count -gt 0) {
+                $completedJobs = $jobs | Where-Object {($_.State -ieq "Completed") -or ($_.State -ieq "Failed")}
+                if (($completedJobs | Measure-Object).Count -gt 0) {
+                    # Write output for completed jobs, but one by one so output doesn't bleed 
+                    # together, also use "Continue" so we write the error but don't end the outer script
+                    $completedJobs | ForEach-Object {
+                        $_ | Receive-Job -ErrorAction Continue
+                    }
+                    # Trim off the completed jobs from our list of jobs
+                    $jobs = $jobs | Where-Object {$_.Id -notin $completedJobs.Id}
+                    # Remove the completed jobs from memory
+                    $completedJobs | Remove-Job
+                }
+                # Wait for 60 sec before checking job status again
+                Start-Sleep -Seconds 60
+            }
+        }
+
+
+        # Needs to create resources in this order, aka parallelize in these three groups, otherwise we get contentions:
+        # i.e. different jobs trying to create the same common resource (RG or lab account)
+        New-ResourceGroups  -ConfigObject $aggregateLabs
+        New-Accounts        -ConfigObject $aggregateLabs
+        New-AzLabMultiple   -ConfigObject $aggregateLabs
+    }
+}
+
+function Set-LabProperty {
+    [CmdletBinding()]
+    param(
+        [parameter(Mandatory = $true, HelpMessage = "Array containing one line for each lab to be created", ValueFromPipeline = $true)]
+        [ValidateNotNullOrEmpty()]
+        [psobject[]]
+        $labs,
+
+        [Parameter(Mandatory = $true, ValueFromRemainingArguments=$true, HelpMessage = "Series of multiple -propertyName propValue pairs")]
+        [ValidateNotNullOrEmpty()]
+        [string[]]
+        $vars
+    )
+    begin {
+        #Convert vars to hashtable
+        $htvars = @{}
+        $vars | ForEach-Object {
+            if($_ -match '^-') {
+                #New parameter
+                Write-Verbose $_
+                $lastvar = $_ -replace '^-'
+                $lastvar = $lastvar -replace ':' # passing parameters as hashtable inserts a : char
+                $htvars[$lastvar] = $null
+            } else {
+                #Value
+                $htvars[$lastvar] = $_
+            }
+        }
+    }
+
+    process {
+        foreach ($l in $labs) {
+            # Deep cloning not to change the original
+            $lc = [System.Management.Automation.PSSerializer]::Deserialize(
+                    [System.Management.Automation.PSSerializer]::Serialize($l))
+
+            Write-Verbose ($lc | Out-String) 
+
+            function ChangeLab ($lab) {
+                $htvars.Keys | ForEach-Object { $lab.($_) = $htvars[$_]}
+            }
+            $lc | ForEach-Object { ChangeLab  $_}
+            $lc
```

Comment:
We aren't saving the changes back to Azure, just changing them in the object in the pipeline.  I don't think it would be clear to the user that they would need an additional step of actually saving the changes - especially since there isn't a "Update-Lab" commandlet...